exonum/src/api/error.rs

```diff
@@ -16,10 +16,118 @@
 
 use failure::Fail;
 use serde::Serialize;
+
+use std::collections::HashMap;
 use std::io;
 
 use crate::node::SendError;
 
+/// List of possible HTTP response codes.
+#[derive(Debug, Serialize)]
+pub enum HttpCode {
+    #[allow(missing_docs)]
+    BadRequest = 400,
+    #[allow(missing_docs)]
+    NotFound = 404,
+    #[allow(missing_docs)]
+    NotImplemented = 501,
+}
+
+/// API HTTP error struct.
+#[derive(Fail, Debug, Serialize)]
+pub struct ApiError {
+    /// HTTP error code.
+    #[serde(skip)]
+    pub http_code: HttpCode,
+    /// A URI reference to the documentation or possible solutions for the problem.
+    #[serde(rename = "type")]
+    pub error_type: String,
+    /// Short description of the error.
+    pub title: String,
+    /// Detailed description of the error.
+    pub detail: String,
```

Comment:
Nit: It could make sense to skip serializing empty fields:

```rust
#[serde(default, skip_serializing_if = "String::is_empty")]
pub detail: String,
```